driver-async/src/test/functional/com/mongodb/async/client/InitialDnsSeedlistDiscoveryTest.java

```diff
@@ -77,23 +81,65 @@ public InitialDnsSeedlistDiscoveryTest(final String filename, final String uri,
     }
 
     @Test
-    public void shouldResolve() {
+    public void shouldResolve() throws InterruptedException {
 
         if (isError) {
+            MongoClient client = null;
```

Comment:
This is awkward, and something we can address with the new spec tests that we'll need. The issue is that some of the error conditions are discovered while constructing the ConnectionString or ClusterSettings, while others are not discovered until the driver attempts to resolve the SRV records.  As a result, we now have to construct a MongoClient and assert on ClusterDescription changes (via SDAM monitoring) in order to assert on the remainder of the errors.